swarm_copy/app/database/sql_schemas.py

```diff
@@ -0,0 +1,61 @@
+"""Schemas for the chatbot."""
+
+import datetime
+import uuid
+
+from sqlalchemy import DateTime, ForeignKey, Integer, String
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column, relationship
+
+
+class Base(DeclarativeBase):
+    """Base declarative base for SQLAlchemy."""
+
+
+def uuid_to_str() -> str:
+    """Turn a uuid into a string."""
+    return uuid.uuid4().hex
+
+
+def utc_now() -> datetime.datetime:
+    """Return the utc time."""
+    return datetime.datetime.now(datetime.timezone.utc).replace(tzinfo=None)
+
+
+class Threads(Base):
+    """SQL table for the users thread / conversations."""
+
+    __tablename__ = "threads"
+    thread_id: Mapped[str] = mapped_column(
+        String, primary_key=True, default=uuid_to_str
+    )
+    vlab_id: Mapped[str] = mapped_column(
+        String, default="430108e9-a81d-4b13-b7b6-afca00195908"
+    )  # only default for now !
+    project_id: Mapped[str] = mapped_column(
+        String, default="eff09ea1-be16-47f0-91b6-52a3ea3ee575"
+    )  # only default for now !
+    title: Mapped[str] = mapped_column(String, default="New chat")
+    creation_date: Mapped[datetime.datetime] = mapped_column(DateTime, default=utc_now)
+    update_date: Mapped[datetime.datetime] = mapped_column(DateTime, default=utc_now)
+
+    user_id: Mapped[str] = mapped_column(String, nullable=False)
+    messages: Mapped[list["Messages"]] = relationship(
+        "Messages", back_populates="thread", cascade="all, delete-orphan"
```

Comment:
Have you tested cascade deletion already ? Back then we used to have the following `cascade` arguments:
```json
{"cascade": "delete, save-update, delete-orphan, merge"}
```
Shouldn't we at least add `delete` here ?

Reply:
Fron chatGPT
```txt

Here’s a breakdown of each cascade argument:

delete:

When a parent object is deleted, all related child objects are also deleted from the database. This cascade operation applies only when the parent is deleted explicitly (e.g., session.delete(parent)).
save-update:

This cascade allows changes made to the parent object to propagate to related child objects automatically. For example, if you add a new child to the relationship list in the parent, this child will be saved in the session automatically, even if it hasn’t been added directly to the session.
delete-orphan:

This cascade will delete any child objects that are orphaned, meaning they are removed from the relationship in the parent object. For instance, if a child object is removed from a parent’s relationship list (e.g., parent.children.remove(child)), the child will be deleted from the database automatically.
merge:

The merge cascade enables related objects to be included when using the session.merge() method. When merging an object into the session, this cascade ensures that the child objects are also merged, updating any existing records with the data from the merged instance.
Summary of Usage
The cascade argument {"cascade": "delete, save-update, delete-orphan, merge"} is comprehensive and provides full control over the lifecycle of related child objects. It allows:

Automatic deletion of child objects when the parent is deleted.
Automatic saving of new or updated child objects when they’re added to the parent.
Automatic deletion of child objects if they are removed from the parent relationship.
Merging of child objects along with their parent during session merges.
This setup is beneficial if you want tight control over the relationship’s lifecycle, especially in applications where you need to keep the related objects in sync with the parent’s state.
```
Feel free to pick the ones that you find the most relevant.